internal/core/algorithm/ngt/ngt.go

```diff
@@ -310,11 +328,12 @@ func (n *ngt) Search(vec []float32, size int, epsilon, radius float32) (result [
 		return nil, errors.ErrIncompatibleDimensionSize(len(vec), int(n.dimension))
 	}
 
-	results := C.ngt_create_empty_results(n.ebuf)
+	ebuf := n.GetErrorBuffer()
+	results := C.ngt_create_empty_results(ebuf)
```

Comment:
do you think we can also cache this empty result object?

Reply:
Hmmm, that's an interesting idea, but I think it's better to do it some other time as we have to check many things like search accuracy, RaceCondition, etc.
For ebufs, there is a clear way to destroy, but for result containers, there is no such I/F, so we have to consider how to destroy used buffers.